### DISCUSSION TITLE
how to make the plot of a non-square grid in a jupyter notebook respect the aspect ratio

### DISCUSSION THREAD

**Discussion by ftalbrecht** [pymor/pymor#1123] [Q&A] [answered]
I am trying to plot a non-square 2d `RectGrid` in a jupyter notebook but the plot does not take the aspect ratio into account. Any ideas what I am doing wrong or how to improve this?

I am on pyMOR `2019.2.x`, an MWE is

```python
%matplotlib widget

import numpy as np

from pymor.grids.rect import RectGrid
from pymor.discretizers.fv import NumpyVectorSpace, PatchVisualizer as Visualizer

grid = RectGrid(num_intervals=[5, 1], domain=([0, 0], [5, 1]))
fv_space = NumpyVectorSpace(dim=grid.size(0))
_ = Visualizer(grid, grid.bounding_box(), codim=0).visualize(
    fv_space.from_numpy(np.arange(grid.size(0))), None)
```

which gives a square plot.

**Comment by renefritze** [comment]
First off, I've changed the MPL backend for jupyter substantially since the 2019.2 release, so my comments probably only apply to the current state.

 - There's no explicit figure size set in/via `pymor.discretizers.builtin.gui.jupyter.matplotlib.visualize_patch`. 
 - the Plot object itself contains an axes object created via tripcolor, which calls `ax.autoscale_view()`. 
 - setting the axes' aspect (to 5 in your example) before displaying the plot should IMO do the trick
 - It doesn't though
 
I would now try to change the initial figure size from the default to default scaled by the bounding box.

**Comment by ftalbrecht** [comment, marked as answer]
I addressed this in #1171. If that is incorporated, the above should work. For future reference: the visualizers should now take the `bounding_box` of the `grid` into account, no need to specify it manually.
